test/built-ins/Temporal/Calendar/prototype/dateAdd/throw-range-error-from-ToTemporalDate.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.dateadd
+description: Temporal.Calendar.prototype.dateAdd should throw from ToTemporalDate.
+info: |
+  ...
+  4. Set date to ? ToTemporalDate(date).
+features: [Temporal]
+---*/
+let cal = new Temporal.Calendar("iso8601");
+
+assert.throws(RangeError,
+    () => cal.dateAdd("invalid date string", new Temporal.Duration(1)),
```

Comment:
We try to reduce the amount of code within the `assert.throws` callback in order to limit ambiguity around the source of the exception. In this case, instantiating the Duration outside of the callback will make it easier to see that it isn't directly under test (and it will prevent Type II errors if the Duration constructor happens to throw).